src/util/config/load.ts

```diff
@@ -536,7 +542,11 @@ export async function resolveConfigs(
     process.exit(1);
   }
 
-  if (!hasProviders) {
+  if (
```

Comment:
please add test coverage for this! How are we not throwing during  `invariant(Array.isArray(config.providers), 'providers must be an array');` below?

Reply:
https://github.com/promptfoo/promptfoo/blob/a08fc4e4da6f64e529e9d6350f307a7e59c82f25/src/util/config/load.ts#L498

whereas `hasProviders` is true when the `.length > 0`.

Reply:
Ahah, ` || []` is bad